hdk-rust/src/api.rs

```diff
@@ -849,16 +849,16 @@ pub fn get_links<S: Into<String>>(base: &Address, tag: S) -> ZomeApiResult<GetLi
 }
 
 /// Returns a list of entries from your local source chain, that match a given type.
-/// entry_type_name: Specify type of entry to retrieve
+/// entry_type_names: Specify type of entry(s) to retrieve, as a Vec<String> of 0 or more names.
 /// limit: Max number of entries to retrieve
-pub fn query(entry_type_name: &str, start: u32, limit: u32) -> ZomeApiResult<QueryResult> {
+pub fn query(entry_type_names: Vec<String>, start: u32, limit: u32) -> ZomeApiResult<QueryResult> {
```

Comment:
I'm not so sure about the overall API here. The current `query` function seems a little bit unergonomic. Maybe we could provide a `QueryBuilder`?

Reply:
Perhaps, although I hope querying the local chain remains a pretty simple function.  This just restores the basic needed functionality to support basic usage -- we need to be able to query for a set of EntryType names, in order to execute local chain state validation machinery, and for that we need to be able to pass multiple query strings, and scan for them efficiently; globset::GlobSetBuilder constructs a single efficient regex matcher to do that.  And, we get good, simple and understandable EntryType name "namespacing" using standard filename-like  "globbing" for free, to support organizing the naming of EntryTypes.

Reply:
Overall this looks fabulous as a way to not build a whole query language and get what we need in the short term.  Nice work.  One thing though: I do think that for most cases people will just want to do:
`hdk::query("post", 0, 0)` not `hdk::query( vec!["post".to_string()], 0, 0)`.  So I propose we do the same thing we did for `hdk::get_entry()` which is that at the hdk level we make the most common use case have the normal name, but then have something like `hdk::query_glob()` which uses the vec, and that's what calls through to `query()` in api.rs

I'll give a more thorough review tommorow.

Reply:
You could also say, that that the first argument is `impl From<QueryArgsNames>`:
```rust
pub fn query(query_args: impl From<QueryArgsNames>, start: usize, limit: usize) { ... }
```
And then do the conversion implicitly in the function body.